docs/ref/configuration.md

```diff
@@ -0,0 +1,39 @@
+# Configuration
+
+- Headscale loads its configuration from a YAML file
+- It searches for `config.yaml` in the following paths:
+  - `/etc/headscale`
+  - `$HOME/.headscale`
+  - the current working directory
+- Use the command line flag `-c`, `--config` to load the configuration from a different path
+- Validate the configuration file with: `headscale configtest`
+
+!!! example "Get the [example configuration from the GitHub repository](https://github.com/juanfont/headscale/blob/main/config-example.yaml)"
+
+    Always select the [same GitHub tag](https://github.com/juanfont/headscale/tags) as the released version you use to
+    ensure you have the correct example configuration. The `main` branch might contain unreleased changes.
+
+    === "View on GitHub"
+
+        * Development version: <https://github.com/juanfont/headscale/blob/main/config-example.yaml>
+        * Version 0.23.0: <https://github.com/juanfont/headscale/blob/v0.23.0/config-example.yaml>
```

Comment:
Wonder if there is a way these can be variables so we only have to change them in one place 🤔

Reply:
I have this on my list, could also be useful for admonitions used multiple times. Probably some kind of doc snippets.

Reply:
Sounds good!

Reply:
Initial try using the [mkdocs-macros-plugin](https://mkdocs-macros-plugin.readthedocs.io/): https://github.com/juanfont/headscale/pull/2163/commits/9aa60cb45dcffefcb4678de5999119fa67303d83